src/main/java/school/hei/haapi/repository/dao/UserManagerDao.java

```diff
@@ -87,6 +86,28 @@ public List<User> findByCriteria(
       predicate = builder.and(predicate, builder.equal(root.get("sex"), sex));
     }
 
+    if (WORKING.equals(workStatus)) {
+      predicate =
+          builder.and(
+              predicate,
+              builder.lessThanOrEqualTo(
+                  workDocumentJoin.get("commitmentBegin"), commitmentComparison));
+    }
+    if (HAVE_BEEN_WORKING.equals(workStatus)) {
+      predicate =
+          builder.and(
+              predicate,
+              builder.lessThanOrEqualTo(
+                  workDocumentJoin.get("commitmentEnd"), commitmentComparison));
+    }
+    if (WILL_BE_WORKING.equals(workStatus)) {
+      predicate =
+          builder.and(
+              predicate,
+              builder.greaterThanOrEqualTo(
+                  workDocumentJoin.get("commitmentBegin"), commitmentComparison));
+    }
+
```

Comment:
add a note that you rely on data validation :
* data validation ensures that commitmentBegin <= commitmentEnd hence validating these cases, otherwise, they would be invalid